### ISSUE TITLE
istioctl install  output installed version inconrrect

### ISSUE THREAD

**Issue by nicole-lihui** [istio/istio#53323]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

when i specify install version is 1.21.4 , then we can seen the describe of install version status is istioctl version

![image](https://github.com/user-attachments/assets/da11dd0c-a727-4e14-9e85-2f5b957b6c9e)

![image](https://github.com/user-attachments/assets/b8d8a118-4bfd-4898-9ebb-18dd09bdafaa)

### Version

```Text
$ istioctl version                 
client version: 1.23.0
control plane version: 1.22.4
data plane version: none

$ kubectl version 
Client Version: v1.31.0
Kustomize Version: v5.4.2
Server Version: v1.31.0
```

**Comment by howardjohn** [comment]
This is not incorrect, how you are installing is. The istioctl binary embeds the Kubernetes manifests which are tightly coupled to a version. Overriding the tag to a different version is making an amalgamation of 2 versions of istio which is bound to not work

**Comment by howardjohn** [comment]
to be clear - use istioctl 1.22.4 to install 1.22.4

**Comment by ilrudie** [comment]
maybe the fix is to warn folks about this state? "tag != istioctl version"

**Comment by bleggett** [comment]
`istioctl` should pull the charts from the Istio OCI repo like it does images, and then we can ditch manifest embeds, IMO.

**Comment by nicole-lihui** [comment]
> `istioctl` should pull the charts from the Istio OCI repo like it does images, and then we can ditch manifest embeds, IMO.
> 
> Should be able to fetch and override charts exactly like images in all cases (and would make it easier to move between versions without switching binaries) - this would generally be more intuitive than explaining that `istioctl` embeds a copy of the charts (but only the charts) and the various bad states this can allow you to get into given that chart and image overrides are handled very differently.
> 
> (This may still result in mismatches between `istioctl` and what it expects the manifests to look like - but IMO those assumptions and mismatches are bugs/tight coupling problems we should fix or explicitly treat as a public API contract in `istioctl`/helm/etc)

I also believe that installing with `istioctl` should allow installation from a standard Helm Repo, rather than forcing a coupling between `ctl` and the installation version. 
Recently, Istio has also been promoting standardized Helm installations, which is likely what everyone is hoping for.
